src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -1281,7 +1281,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     -->
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.GetEmbeddedApphostPaths"
           AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
-  <Target Name="ComputeEmbeddedApphostPaths">
+  <Target Name="ComputeEmbeddedApphostPaths" Condition="'$(PackAsTool)' == 'true'">
```

Comment:
wow crazy that we called this all of the time when its tool specific. that target name has no implication that this is tool code only